### PR TITLE
feat: Change how round closes from time based to owner set

### PR DESCRIPTION
**Pull request by jsuresh**
@ashvds @T04435 - I think we need to manually close our presale round as opposed to time based this time around.

Still a few tasks required to bring this home (namely tests). I'll try and finish this off, but if someone else has some time/space/wants to make their first small smart contract change then please, jump in :)

 - [x] Write/update tests

### CODE REVIEW

File: contracts/Presale.sol [hunk 1]
```diff
@@ -39,31 +42,34 @@ contract Presale is Ownable {
...
 
-    function setIssuedToken(IERC20 _issuedToken) external onlyOwner {
-        issuedToken = _issuedToken;
-    }
-
     function depositFor(address account, uint256 amount) external {
         require(account != address(0), "Presale: Address cannot be 0x0");
-        require(block.timestamp < roundEndTimestamp, "Presale: round closed");
+        require(isOpen, "Presale: Round closed");
+        require(hardCap > totalAllocated, "Presale: Round closed, hard cap reached");
```
Comment (ashvds) [inline contracts/Presale.sol:61]: @jsuresh can we have this be "goal reached" instead of hard cap?

Comment (ashvds) [inline contracts/Presale.sol:61]:
```suggestion
        require(hardCap > totalAllocated, "Presale: Round closed, goal reached");
```

File: test/presale-tests.ts [hunk 2]
```diff
@@ -79,9 +84,21 @@ describe('Dogfood Presale Tests', function () {
...
+      await presale.closeRound();
       await expect(presale.depositFor(await ash.getAddress(), toAtto(100)))
-        .to.revertedWith("Presale: round closed");
+        .to.revertedWith("Presale: Round closed");
+    });
+
+    it('should block deposits, once round hard cap is reached', async function () {
+      await presale.depositFor(await jeeva.getAddress(), toAtto(1000));
+      await expect(presale.depositFor(await ash.getAddress(), toAtto(100)))
+        .to.revertedWith("Presale: Round closed, hard cap reached");
```
Comment (ashvds) [inline test/presale-tests.ts:95]:
```suggestion
        .to.revertedWith("Presale: Round closed, goal reached");
```

**Review by ashvds — APPROVED** [review]
LGTM%

File: contracts/Presale.sol (hunk 1 above)
Comment (jsuresh) [reply contracts/Presale.sol:61]: Created issue to do as a follow up task (simple string change + update of tests): https://github.com/saftely/saftely-protocol/issues/5

Comment (jsuresh) [reply contracts/Presale.sol:61]: @samfroggy - prob want to stop the dapp allowing people to ape in if the round is closed or we hit our goal to save our friends some gas :)

**Review by ashvds — APPROVED** [review]

**Pull request merged**